File: sdks/js/bench/emit.bench.ts
import { bench, describe } from 'vitest';
import { createLoza, production, memorySink } from '../src';

describe('emit', () => {
  const logger = createLoza(production('bench').withSink(memorySink()));

  bench('emit cycle', () => {
    const ctx = logger.startEvent({ event: 'bench.test', kind: 'http' });
    logger.finish(ctx, 'success');
    logger.emit(ctx);
  });
});

File: sdks/rs/bench/emit_bench.rs
use loza::{Config, New, Params, String as LozaString};

pub fn run_emit_iterations(iterations: usize) -> usize {
    let logger = New(Config::test("bench"));
    let mut bytes = 0;
    for idx in 0..iterations {
        let mut ctx = logger.start_event(Params::new("bench.emit"));
        logger.append(&mut ctx, LozaString("iteration", &idx.to_string()));
        logger.finish(&mut ctx, "success");
        bytes += logger.emit(&ctx).unwrap_or_default().len();
    }
    bytes
}

